src/common/searchpath.c

```diff
@@ -296,10 +296,23 @@ void AddSubSearchPathFromBin (SearchPaths* P, const char* SubDir)
     }
     *Ptr = '\0';
 
-#else
+#elif defined(__linux__)
```

Comment:
this isn't necessary; in fact it's just cargo culting. 2 years ago when i needed the functionality here i came across the stackoverflow answer mentioning this method, i looked at it in digust and instead went to find a portable method - the one presented in this PR.
this addition here will make not only add a method that works only on a single platform, whereas the other one works everywhere, it also relies on /proc being mounted - which isn't always the case. so it works less reliable, i.e. linux is now at a disadvantage vs e.g. HAIKU or the BSDs.

Reply:
oh ok, i didn't notice that GetProgPath is still called as a fallback, so i retract the latter.

Reply:
The fallback is always needed as a last resort - however to make that edgy usecase work, using argv[0] will not do the trick

Reply:
> using argv[0] will not do the trick

why ?

Reply:
oh, i see this was added to make `make avail` work, which i didnt study yet. i'll study it in a couple hours, though i'm quite sure it will work even without the /proc hack.

Reply:
Look at what the "make avail" target does.... it creates symlinks to (only) the binaries in "/usr/local/bin", nothing else is installed. Now when PATH is pointing to /usr/local/bin and you run the binary as "cc65", the code would add "/usr/local/" to the searchpath - but there isnt anything. The binary could still be located in /home/user/cc65/bin and the includes and libs in /home/user/cc65/

Reply:
> oh, i see this was added to make `make avail` work, which i didnt study yet. i'll study it in a couple hours, though i'm quite sure it will work even without the /proc hack.

As i said before, the only way to make it work is adding some more symlinks (to the lib/include/asminc/cfg dirs) and then compile with PREFIX=/usr/local - but it'd rather not do this, its even more ugly than what it does now. The proc lookup is fine and works for all practical purposes. If you can make it work without doing either - i'd really like to know how :)

Reply:
without having studied the issue in much detail, i'd think realpath() would solve even the case where "absolute path involving symlinks" is involved.

from man 3p realpath()

```
     The  realpath()  function  shall  derive, from the pathname pointed to by
     file_name, an absolute pathname  that  resolves  to  the  same  directory
     entry,  whose  resolution  does not involve '.', '..', or symbolic links.
```

anyway, i'll be afk for a couple hours, then will get back to this and see whether we can find an elegant solution.

Reply:
Some quick testing seems to suggest the solution could be:
1) make GetProgPath() always return the path including the binary name
2) call realpath on the returned path
3) strip the binary name from it

Interesting!

That said, we'll still need some ifdef magic around 2) - because not everything that isnt windows is POSIX (AmigaOS for example)
Also: the

Reply:
i resolved it slightly different - GetProgPath now always passes through realpath(), so we don't need to call it twice in the case of starting the binary using a relative path, like bin/cc65.
but it does return the full binary name now, so we can use the previous existing code to strip it, just like the windows code. please take a look at the last commit.